data/src/main/java/com/microsoft/azure/kusto/data/Utils.java

```diff
@@ -162,11 +162,13 @@ private static DataServiceException createExceptionFromResponse(String url, Http
                     return new DataServiceException(url, message, formattedException,
                             formattedException.getApiError().isPermanent());
                 } catch (Exception ignored) {
+                    // Exception doesn't follow the OneAPI error format
+                    return new DataServiceException(url, errorFromResponse, false);
                 }
             }
             errorFromResponse = String.format("Http StatusCode='%s', ActivityId='%s'", httpResponse.getStatusLine().toString(), activityId);
             if (thrownException == null) {
-                thrownException = new DataWebException(errorFromResponse, httpResponse);
+                return new DataServiceException(url, errorFromResponse, false);
```

Comment:
Don't we miss the Http code here?

Reply:
I changed the hierarchy (in a non-breaking way), now there is a `WebException` class for exceptions that aren't oneapi, and there's an api from DataServiceException to get the error code, if available.